src/impl_/pyclass.rs

```diff
@@ -189,6 +189,204 @@ pub trait PyClassImpl: Sized {
     }
 }
 
+/// Iterator used to process all class items during type instantiation.
+pub struct PyClassItemsIter {
```

Comment:
Would it make sense to implement `size_hint` and/or `ExactSizeIterator` or do we rely on inlining and code expansion for performance anyway?

Reply:
I suspect that in this case it might be inlined anyway as the only code which is expected to be iterating this type is the `create_type_object` function inside of PyO3.

I don't think size_hint / `ExactSizeIterator` is likely to be relevant as that's usually (afaik) for deciding how much space to allocate in a collection as part of a call to `.collect()`? Also in `multiple-pymethods` case I'm not sure whether we can support it cleanly (doesn't look like `inventory::iter` struct implements these hints).

Reply:
Perhaps worth pointing out I'm not overly concerned about maximising performance here as this will be part of one-time setup for pyclasses on module import and never executed again after that. As long as we're not doing anything terrible, perf isn't likely to be noticeable here.